Sources/firebase/include/FirebaseFirestore.hh

```diff
@@ -44,6 +44,37 @@ document_get(const ::firebase::firestore::DocumentReference document, ::firebase
   return document.Get(source);
 }
 
+inline ::firebase::firestore::DocumentReference
+snapshot_reference(const ::firebase::firestore::DocumentSnapshot snapshot) {
+  return snapshot.reference();
+}
+
+struct MapFieldValue_Workaround {
+  std::vector<std::string> keys;
+  std::vector<::firebase::firestore::FieldValue> values;
+};
```

Comment:
This type and the resulting _workaround apis are here because it seems that CxxDictionary conformance doesn't work well in Swift for Windows yet. I have a sample repo that shows the problem and need to file an issue on the Swift repo https://github.com/brianmichel/swift-cpp-interop-map-compatibility.

These are kind of ugly, but the MapFieldValue_Workaround type just breaks out the keys/values of a regular map into accessible vectors.